### ISSUE TITLE
IconButton needing an aria-label 

### ISSUE THREAD

**Issue by divyajindel** [Sage/carbon#4552]
### Current behaviour

IconButton needing an aria-label because it doesn’t have any discernible text[
<img width="1117" alt="Screenshot 2021-11-10 at 16 41 02" src="https://user-images.githubusercontent.com/88449494/141156228-9029a404-2504-4bea-9fba-8c5f9552cab6.png">
](url)

### Expected behaviour

it should have discernible text with No axe Violation.

### CodeSandbox or Storybook URL

https://carbon.sage.com/?path=/story/icon-button--default-story

### Carbon Version

latest

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Microsoft Edge

### What Operating System are you seeing the problem on?

MacOS

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by sianford** [comment]
FE-4466
